lib/matplotlib/pyplot.py

```diff
@@ -2164,16 +2296,18 @@ def set_cmap(cmap):
 
 
 @_copy_docstring_and_deprecators(matplotlib.image.imread)
-def imread(fname, format=None):
+def imread(fname: str | io.FileIO, format: str | None = None) -> np.ndarray:
```

Comment:
Should have `os.PathLike` as well?

Reply:
Possibly, I think it may work.

That said, the docstring is listed as `str or file-like` and there are `isinstance(fname, str)` checks in the implementation, so I kept it as the narrower type.

My inclination is to leave it for now, but maybe do a narrower change that enables any `os.PathLike` and ensures that the implementation matches, the docstring is updated, and the type hint is updated both in `image.pyi` and in `pyplot.py` (this is not an autogenerated method)

Reply:
we should support PathLike everythere (even if the docs did not catch up).

Reply:
PIL doesn't actually cover the full `os.PathLike` (though the common cases of `bytes`, `str`, and `pathlib.Path` are covered)

They use the following check:

https://github.com/python-pillow/Pillow/blob/583dd3b4245649ddd367c2dcbbf2bf9fb7835017/src/PIL/_util.py#L5-L6